### ISSUE TITLE
No way to use BASIC on the Gigatron microcomputer

### ISSUE THREAD

**Issue by Epictyphlosion** [mamedev/mame#7145]
Me again. I noticed that the Gigatron microcomputer got promoted to working in 0.224, which is pretty cool. However, there's currently no way to input anything from the keyboard, which makes it impossible to use BASIC or WozMon.

I guess this is kind of a weird case, since making it keyboard input would mean it would have to be treated like other microcomputers in MAME, but then that would make all the other programs in the machine useless. Not to mention that you have to hook up an adapter to it's controller port for it to accept keyboard input in the first place. Dunno if something like that would be possible to emulate

**Comment by Sterophonick** [comment]
Yeah, this is a goal of mine for the driver. The keyboard is implemented kinda funny compared to other computers.
It's definitely possible to HLE the ATtiny85 that's on Pluggy McPlugface, but I don't know if the MAME team will allow me to do that.

**Comment by rb6502** [comment]
Right, you'd want to implement this the same way as on the real machine, by having pluggable options for the controller port, one of which is the keyboard/adapter.

If the ATtiny85 code is available, we'd definitely prefer it be emulated.
